marathon_lb.py

```diff
@@ -646,12 +647,32 @@ def reloadConfig():
         logger.info("reloading using %s", " ".join(reloadCommand))
         try:
             start_time = time.time()
+            checkpoint_time = start_time
+            # Retry or log the reload every 10 seconds
+            reload_frequency = args.reload_interval
             old_pids = get_haproxy_pids()
             subprocess.check_call(reloadCommand, close_fds=True)
+            new_pids = get_haproxy_pids()
+            logger.debug("Waiting for new haproxy pid (old pids: [%s], " +
+                         "new_pids: [%s])...", old_pids, new_pids)
             # Wait until the reload actually occurs and there's a new PID
-            while len(get_haproxy_pids() - old_pids) < 1:
-                logger.debug("Waiting for new haproxy pid...")
+            while True:
+                new_pids = get_haproxy_pids()
```

Comment:
suggestion: move this to the end of the while loop to eliminate a couple of redundant calls